plugins/integrations/kubernetes-service/src/main/java/com/cloud/kubernetes/cluster/KubernetesClusterManagerImpl.java

```diff
@@ -379,8 +380,8 @@ public VMTemplateVO getKubernetesServiceTemplate(DataCenter dataCenter, Hypervis
     protected void validateIsolatedNetworkIpRules(long ipId, FirewallRule.Purpose purpose, Network network, int clusterTotalNodeCount) {
         List<FirewallRuleVO> rules = firewallRulesDao.listByIpAndPurposeAndNotRevoked(ipId, purpose);
         for (FirewallRuleVO rule : rules) {
-            Integer startPort = rule.getSourcePortStart();
-            Integer endPort = rule.getSourcePortEnd();
+            int startPort = ObjectUtils.defaultIfNull(rule.getSourcePortStart(), 1);
+            int endPort = ObjectUtils.defaultIfNull(rule.getSourcePortEnd(), KubernetesClusterActionWorker.MAX_PORT);
```

Comment:
good catch @GaOrtiga 

should firewall with other protocols (not TCP) be skipped ?

Reply:
Thank you @weizhouapache.

Can you further explain this comment?

As it stands this PR maintains the current behaviour of validating rule conflicts for rules with every protocol, without any differentiation between TCP, UDP and ICMP. I have, however, run tests using all three protocols, and all of them shared the same results.

Reply:
@GaOrtiga 
this code snippet checks if there are firewall rules conflicting with the kubernetes API (tcp/6443) and node ssh (tcp/2222, tcp/2223, etc).
IMHO, if firewall rules do not use TCP protocol, the ports check can be skipped. for instance udp/6443 has no conflicts with tcp/6443.

I can create another pr for it. never mind.